From 535a40ff57fa3fe3ef5ca339ac30de4754729093 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Mon, 28 Nov 2016 13:27:42 +0200
Subject: [PATCH 0627/1345] fix: dts: a80x0: move board specific configuration
 to dts

commit  243dc8f9a7c332b0387eefed3ec4c4de8b3e62e5 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- move specific configuration of router to be in board dts file,
instead of in shared SoC dtsi file.
- disable I2C and SPI on AP for armada-8040-db-A configuration.
  (both I2C & SPI are not connected on AP on A1 board)

Change-Id: I22ba030e6ac81423e45f949afcb28d1409798b19
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34129
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts   |   18 ++++++-
 .../boot/dts/marvell/armada-8040-db-router.dts     |   46 ++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi    |   55 ++------------------
 3 files changed, 66 insertions(+), 53 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
index d9bfeb6..8e9afaa 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
@@ -56,8 +56,22 @@
 
 	chosen { };
 
-	cpn-110-master {
-		config-space {
+	ap806 {
+		config-space@f0000000 {
+			serial@512000 {
+				status = "okay";
+			};
+			i2c@511000 {
+				status = "disabled";
+			};
+			spi0: spi@510600 {
+				status = "disabled";
+			};
+		};
+	};
+
+	cp110-master {
+		config-space@f2000000 {
 			sata@540000 {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
index 3f813d4..2e6e9d6 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
@@ -56,6 +56,52 @@
 
 	chosen { };
 
+	ap806 {
+		config-space@f0000000 {
+			serial@512000 {
+				status = "okay";
+			};
+			i2c@511000 {
+				status = "okay";
+				clock-frequency = <100000>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&ap_i2c0_pins>;
+				eeprom@57 {
+					compatible = "at24,24c64";
+					reg = <0x57>;
+					pagesize = <32>;
+				};
+				eeprom@50 {
+					compatible = "at24,24c64";
+					reg = <0x50>;
+					pagesize = <32>;
+				};
+			};
+			spi0: spi@510600 {
+				status = "okay";
+				spi-flash@0 {
+					#address-cells = <0x1>;
+					#size-cells = <0x1>;
+					compatible = "jedec,spi-nor";
+					reg = <0x0>;
+					spi-max-frequency = <20000000>;
+					partition@0 {
+						label = "boot";
+						reg = <0x0 0x200000>;
+					};
+					partition@200000 {
+						label = "Filesystem";
+						reg = <0x200000 0xd00000>;
+					};
+					partition@d00000 {
+						label = "boot_2nd";
+						reg = <0xf00000 0x100000>;
+					};
+				};
+			};
+		};
+	};
+
 	cp110-master {
 		config-space@f2000000 {
 			sata@540000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 7d67961..1bba048 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -56,55 +56,8 @@
 		reg = <0x0 0x0 0x0 0x80000000>;
 	};
 
-	ap806 {
-		config-space@f0000000 {
-			serial@512000 {
-				status = "okay";
-			};
-			i2c@511000 {
-				status = "okay";
-				clock-frequency = <100000>;
-				pinctrl-names = "default";
-				pinctrl-0 = <&ap_i2c0_pins>;
-				eeprom@57 {
-					compatible = "at24,24c64";
-					reg = <0x57>;
-					pagesize = <32>;
-				};
-				eeprom@50 {
-					compatible = "at24,24c64";
-					reg = <0x50>;
-					pagesize = <32>;
-				};
-			};
-			spi0: spi@510600 {
-				status = "okay";
-				spi-flash@0 {
-					#address-cells = <0x1>;
-					#size-cells = <0x1>;
-					compatible = "jedec,spi-nor";
-					reg = <0x0>;
-					spi-max-frequency = <20000000>;
-					partition@0 {
-						label = "boot";
-						reg = <0x0 0x200000>;
-					};
-					partition@200000 {
-						label = "Filesystem";
-						reg = <0x200000 0xd00000>;
-					};
-					partition@d00000 {
-						label = "boot_2nd";
-						reg = <0xf00000 0x100000>;
-					};
-				};
-			};
-
-		};
-	};
-
-	cpn-110-master {
-		config-space {
+	cp110-master {
+		config-space@f2000000 {
 			i2c@701000 {
 				status = "okay";
 				clock-frequency = <100000>;
@@ -186,8 +139,8 @@
 		};
 	};
 
-	cpn-110-slave {
-		config-space {
+	cp110-slave {
+		config-space@f4000000 {
 			i2c@701000 {
 				eeprom@57 {
 					compatible = "at24,24c64";
-- 
1.7.9.5

